src/bosh-director/lib/bosh/director/deployment_plan/availability_zone.rb

```diff
@@ -3,14 +3,23 @@ module DeploymentPlan
     class AvailabilityZone
       extend ValidationHelper
 
-      def self.parse(availability_zone_spec)
+      def self.parse(cloud_config)
```

Comment:
I think this refactor introduces more confusion than it's worth. With a `parse` method on a singular class, I expect it to receive input to return me a singular object. I also don't like that we're moving in top-level, cloud-config `azs` type configuration data here since that should be out of scope of this naive model.

I don't think this is the correct level for this refactoring. Should it be a class method of `CloudManifestParser ` instead? Unless I'm missing something, mind re-refactoring this?